src/uct/ib/base/ib_iface.c

```diff
@@ -1286,6 +1286,41 @@ uct_ib_iface_init_roce_addr_prefix(uct_ib_iface_t *iface,
     return UCS_OK;
 }
 
+static unsigned uct_ib_iface_gid_index(uct_ib_iface_t *iface)
+{
+    uct_ib_device_t *dev = uct_ib_iface_device(iface);
+    uint8_t port_num     = iface->config.port_num;
+    int gid_tbl_len      = uct_ib_device_port_attr(dev, port_num)->gid_tbl_len;
+    unsigned gid_index   = UCT_IB_MD_DEFAULT_GID_INDEX;
+    uct_ib_device_gid_info_t gid_info;
+    ucs_status_t status;
+    uint32_t oui;
+
+    if (!iface->config.flid_enabled ||
+        (gid_tbl_len <= UCT_IB_MD_DEFAULT_ROUTE_GID_INDEX)) {
+        goto out;
+    }
+
+    status = uct_ib_device_query_gid_info(
+                          dev->ibv_context, uct_ib_device_name(dev), port_num,
+                          UCT_IB_MD_DEFAULT_ROUTE_GID_INDEX, &gid_info);
```

Comment:
shall we search all valid GIDs, not only index 1?

Reply:
afaiu, it is not necessary.
need to change?

Reply:
ok, so maybe rename to UCT_IB_DEVICE_FLID_GID_INDEX?

Reply:
ok

Reply:
btw, i'd make it `UCT_IB_MD_DEFAULT_FLID_ROUTE_GID_INDEX` then, because when FLID routing is configured, all GIDS contain FLID, but we have to mention that we need routable GID

Reply:
UCT_IB_DEVICE_ROUTABLE_FLID_GID_INDEX

Reply:
shall we rename `UCT_IB_MD_DEFAULT_GID_INDEX` then? Otherwise we would have different name patterns for different gid indexes. So either we have 
```
UCT_IB_MD_DEFAULT_GID_INDEX
UCT_IB_MD_ROUTABLE_FLID_GID_INDEX
```
or
```
UCT_IB_DEVICE_DEFAULT_GID_INDEX
UCT_IB_DEVICE_ROUTABLE_FLID_GID_INDEX
```

Reply:
IMO 2nd option, and move it to ib_device.h